notebooks/15_flightplanning_tuto.ipynb

```diff
@@ -0,0 +1,1018 @@
+    "warnings.filterwarnings(\"ignore\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## About the Flight Planning problem"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this problem, an aircarft is going from a departure airport to a destination airport. If the weather is not taken into account the optimal (shortest) path would be the [great circle](https://en.wikipedia.org/wiki/Great_circle). When winds come into play the prediction of the trajectory is more complex as we need to take into account the speed of the wind which varies over time. In the real life an aircraft route is defined by a certain number of waypoint the aircraft must go through and flight levels the pilot is supposed to respect. In order to simplify the problem, we will assume our aircraft can follow the directed edges of a graph starting from the destination airport and finishing to the arrival airport."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Taming weather data \n",
+    "\n",
+    "As discussed earlier we want to see the impact of weather and more specifically wind on the flight path. Therefore we neeed to be able to get the wind velocity for any point of the globe at any flyable altitude and for a given time.\n",
+    "\n",
+    "For this we will use utility functions that are defined in `flightplanning_utils.py`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining credentials"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The weather data will be retrieved from the [Climate Data Store](https://cds.climate.copernicus.eu/cdsapp#!/home). To enable retrieval you will need to create an account there and store your credentials as follows:\n",
+    "\n",
+    "```\n",
+    "key=\"{your_id}:{your_key}\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "key = \"123456:aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieving wind data\n",
+    "\n",
+    "The next cell will retrieve the wind information for a given day on the whole planet. This will download approximately `2 GB`of data. Make sure to have a proper connection."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "if key != \"123456:aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee\":\n",
+    "    wr = WeatherRetrieverFromEcmwf()\n",
+    "\n",
+    "    # Select a date in the past, it takes up to 7 days for the weather data to reach the server\n",
+    "    now = datetime.now()\n",
+    "\n",
+    "    # Here we fetch the data 7 days in the past from now\n",
+    "    file = wr.get(now - timedelta(days=7))\n",
```

Comment:
doesn't work today but probably a copernicus network issue. Maybe we should anticipate this issue and have a code running even with no wind ?

Reply:
Ok will provide a no wind run when the download does work

Reply:
Ok will make a change so that the notebook runs without key (i.e. without wind)